src/event.c

```diff
@@ -148,43 +202,69 @@ int event_check_value(knot_event event, knot_value_type current_val,
 			value_type > KNOT_VALUE_TYPE_MAX)
 		return -EINVAL;
 
-	if (is_change_flag_set(event.event_flags) &&
-			!is_value_equal(current_val, sent_val, value_type))
-		rc = 1;
-	else if (is_lower_flag_set(event.event_flags) &&
-			is_lower_than_threshold(current_val,
+	if (event.event_flags) {
+		switch (event.event_flags) {
+		case KNOT_EVT_FLAG_CHANGE:
+			if (!is_value_equal(current_val, sent_val, value_type))
+				rc = 0;
+			break;
+		case KNOT_EVT_FLAG_LOWER_THRESHOLD:
+			if (!is_lower_than_threshold(current_val,
 				event.lower_limit, value_type))
-		rc = 1;
-	else if (is_upper_flag_set(event.event_flags) &&
-			is_higher_than_threshold(current_val,
+				rc = 0;
+			break;
+		case KNOT_EVT_FLAG_UPPER_THRESHOLD:
+			if (!is_higher_than_threshold(current_val,
 				event.upper_limit, value_type))
-		rc = 1;
-	else
+				rc = 0;
+			break;
+		case KNOT_EVENT_CHANGE_UPPER_FLAG:
+			if (!comp_change_upper_flag(current_val, sent_val,
+						   event.upper_limit,
+						   value_type))
+				rc = 0;
+			break;
+		case KNOT_EVENT_CHANGE_LOWER_FLAG:
+			if (!comp_change_lower_flag(current_val, sent_val,
+				event.lower_limit,
+				value_type))
+				rc = 0;
+			break;
+		case KNOT_EVENT_UPPER_LOWER_FLAG:
+			if (!comp_upper_lower_flag(current_val,
+						event.lower_limit,
+						event.upper_limit,
+						value_type))
+				rc = 0;
+			break;
+		case KNOT_EVENT_FLAG_MAX:
+			if (!comp_all_flag(current_val,
+					event.lower_limit,
+					event.upper_limit,
+					sent_val,
+					value_type))
+				rc = 0;
+			break;
+		default:
+			rc = -1;
+			break;
+		}
+	} else
 		rc = 0;
 
 	return rc;
 }
 
-void event_add_data_item(int id, knot_event event)
-{
-	struct data_item_timeout *data = l_new(struct data_item_timeout, 1);
-
-	data->id = id;
-	data->timeout_sec = event.time_sec;
-
-	if (is_timeout_flag_set(event.event_flags)) {
-		struct l_timeout *to = l_timeout_create(event.time_sec,
-							on_sensor_to,
-							data,
-							l_free);
-		l_queue_push_head(sensor_timeouts, to);
-	}
-}
-
-int event_start(timeout_cb_t cb)
+int event_start(timeout_cb_t cb, int time_sec)
```

Comment:
what does cb means?

Reply:
Callback Function. It is a common way of abbreviating.